samples/matter/closure/src/persistent_percent.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "persistent_percent.h"
+
+#include <cstdlib>
+#include <cstring>
+#include <persistent_storage/persistent_storage.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(persistent_percent, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::DeviceLayer;
+
+PersistentPercent::PersistentPercent(const char *key, uint16_t defaultValue) : mKey(key), mValue(defaultValue) {}
+
+CHIP_ERROR PersistentPercent::Set(uint16_t newValue, bool forceStore)
+{
+	mValue = newValue;
+	uint16_t rounded = RoundToStep(newValue);
+	if (forceStore) {
+		ReturnErrorOnFailure(Store(mValue));
+	} else if (rounded != mStored) {
+		ReturnErrorOnFailure(Store(rounded));
+	}
+	return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentPercent::Store(uint16_t val)
+{
+	Nrf::PersistentStorageNode node{ mKey.c_str(), mKey.size() + 1 };
+	LOG_DBG("Storing %u", val);
+	auto err = Nrf::GetPersistentStorage().NonSecureStore(&node, &val, sizeof(val));
+	if (err == Nrf::PSErrorCode::Success) {
+		mStored = val;
+		return CHIP_NO_ERROR;
+	}
+	return CHIP_ERROR_PERSISTED_STORAGE_FAILED;
+}
+
+CHIP_ERROR PersistentPercent::Load()
+{
+	Nrf::PersistentStorageNode node{ mKey.c_str(), mKey.size() + 1 };
+
+	size_t outSize = 0;
+	auto err = Nrf::GetPersistentStorage().NonSecureLoad(&node, &mValue, sizeof(mValue), outSize);
+	if (err == Nrf::PSErrorCode::Success) {
+		mStored = mValue;
+		return CHIP_NO_ERROR;
+	}
+	// Not found, store default
+	return Store(mValue);
+}
+
+uint16_t PersistentPercent::RoundToStep(uint16_t v)
+{
+	static constexpr uint16_t allowed[] = { 0, 2000, 4000, 6000, 8000, 10000 };
```

Comment:
I guess it can be moved to the class, because you refer to this there in the class description.